### ISSUE TITLE
Left aligned title for SlicerAppBar and FlexibleSpace

### ISSUE THREAD

**Issue by balazsgerlei** [flutter/flutter#11951]
Hi!

I'm trying to implement a "Collapsing Toolbar", or the "Flexible space with image" scrolling technique from the Material Design spec, but I don't have a back button or navigation drawer. So I would like the title to be near the left edge, without any left padding/space. When I use the SliverAppBar and FlexibleSpace I can only get it to display the title with a considerable left padding, which when collapsed gives space to a back button or navigation drawer (hamburger) icon. This is not the behaviour I want, I would like to do something similar to the Android Contacts app:

![example1](http://cdn04.androidauthority.net/wp-content/uploads/2015/12/Contacts-1.png)

Is this possible (without implementing my new SicerAppBar and FlexibleSpace class from scratch), and if so, how to do it?

**Comment by Hixie** [comment]
cc @HansMuller

**Comment by 1and1get2** [comment]
this is brought up a year ago, and I'd still very much like to see this

**Comment by Unfreeze-kk** [comment]
Any solution？

**Comment by crushman1** [comment]
In `FlexibleSpaceBar` set `titlePadding: EdgeInsets.all(0.0)`,

**Comment by hardypatel30** [comment]
@crushman1  this solution would overlap title and left icon in the appbar. We need to find another solution for the same.

**Comment by hardypatel30** [comment]
@Unfreeze-kk ,  @balazsgerlei    Found the solution on my own and added answer there as well.  

https://stackoverflow.com/questions/59243525/sliver-appbar-collapsing-toolbar-animate-title-from-left-to-center-in-flutter/59434248#59434248

Check it.

**Comment by xJon** [comment]
@crushman1's answer actually fits the request of @balazsgerlei, but a more flexible option is possible with a scroll controller: 
![](https://i.stack.imgur.com/L3lOk.gif)
https://stackoverflow.com/a/62137478/7885980

**Comment by jinfagang** [comment]
@Hixie  @xJon Any standard solution to get the exactly same effect?

**Comment by xJon** [comment]
> @Hixie @xJon Any standard solution to get the exactly same effect?

See my answer over https://stackoverflow.com/a/62137478

**Comment by jinfagang** [comment]
@xJon  I tried your anwser, it's nto work like this, not smooth..

**Comment by prazeblaze** [comment]
> > @Hixie @xJon Any standard solution to get the exactly same effect?
> 
> See my answer over https://stackoverflow.com/a/62137478

Unfortunately, this answer does not work for SliverAppBar with kExpandedHeight of less than around 250 value.

**Comment by jinfagang** [comment]
@prazeblaze Do u have a better solution on this? my kExpandedHeight actually less than 250, (100)

**Comment by percula** [comment]
> @prazeblaze Do u have a better solution on this? my kExpandedHeight actually less than 250, (100)

Checkout my solution at https://stackoverflow.com/a/64583870/6591585. It only rebuilds what is necessary and will work for smaller expanded heights.

**Comment by jinfagang** [comment]
@percula Do u have gif to post?

**Comment by percula** [comment]
No gif, just copy/paste. 

I ended up [creating a better solution](https://gist.github.com/percula/dd512e6665835e90a509cd89af8db726) that utilizes the transformations already happening within the FlexibleSpaceBar. Just provide a `titlePaddingTween` such as 
```dart
titlePaddingTween: EdgeInsetsTween(begin: EdgeInsets.only(left: 16.0, bottom: 16), end: EdgeInsets.only(left: 72.0, bottom: 16))
```
instead of `titlePadding`. 

I also added a `foreground` parameter that displays above the `title` and `background`, but doesn't transform as they do.
